src/test/fuzz/versionbits.cpp

```diff
@@ -276,47 +261,51 @@ FUZZ_TARGET_INIT(versionbits, initialize)
 
     // since is straightforward:
     assert(since % period == 0);
-    assert(0 <= since && since <= current_block->nHeight + 1);
+    assert(0 <= since && since <= height);
     if (state == exp_state) {
         assert(since == exp_since);
     } else {
-        assert(since == current_block->nHeight + 1);
+        assert(since == height);
     }
 
     // state is where everything interesting is
     switch (state) {
     case ThresholdState::DEFINED:
         assert(since == 0);
         assert(exp_state == ThresholdState::DEFINED);
-        assert(current_block->GetMedianTimePast() < checker.m_begin);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
+        assert(never_active_test || height < checker.m_begin);
+        assert(never_active_test || height < checker.m_end);
```

Comment:
I don't quite follow as to why the range for both startheight and timeoutheight should be ` [0, period*(max_periods-2)]`. I had set the range currently used in order to avoid the issue discussed in https://github.com/bitcoin/bitcoin/pull/21392#discussion_r597258950.

Reply:
For bip9/mtp versionbits the cache lookup loop will say "oh, we haven't reach start time yet, must still be DEFINED" which is inconsistent with the "DEFINED" check afterwards that says "if we reached timeout and start time at the same time, we go to FAILED instead of STARTED". If the fuzzer tracked the mtp for the previous period, you could capture that logic too, but that seemed a bit messy.

Because bip8 versionbits always goes DEFINED to STARTED it doesn't have that inconsistency, so checking a broader range of parameters is easy.

Reply:
Done